torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -686,6 +686,34 @@ bool ProcessGroupNCCL::WorkNCCL::checkTimeout(
       " milliseconds before timing out.");
 
   LOG(ERROR) << exceptionMsg;
+
+  // Get the stack trace of the work at call time
+  // First step we get the corresponding record entry from FR, based on work's
+  // trace_id_
+  std::optional<NCCLTraceBuffer::Entry> entry =
+      NCCLTraceBuffer::get()->getEntry(trace_id_);
+  if (entry.has_value()) {
+    auto entryVal = entry.value();
+    // Get stack trace from FR entry, in string format
+    // Note: `getTraceback` call below invokes `torch::symbolize`.
+    // According to @fduwjj, `torch::symbolize` may need to acquire the GIL. In
```

Comment:
it's not according to me, it's according to the code. Shall we not write a comment like this?

Reply:
Maybe you can just say "`torch::symbolize` would acquire the GIL, so we add a timeout to make it non-blocking" something like that?

Reply:
Removed, sorry